autogpt_platform/backend/backend/data/sqlalchemy.py

```diff
@@ -0,0 +1,233 @@
+"""
+SQLAlchemy infrastructure for AutoGPT Platform.
+
+This module provides:
+1. Async engine creation with connection pooling
+2. Session factory for dependency injection
+3. Database lifecycle management
+"""
+
+import logging
+import re
+from contextlib import asynccontextmanager
+from typing import AsyncGenerator
+
+from sqlalchemy.ext.asyncio import (
+    AsyncEngine,
+    AsyncSession,
+    async_sessionmaker,
+    create_async_engine,
+)
+
+from backend.util.settings import Config
+
+logger = logging.getLogger(__name__)
+
+# ============================================================================
+# CONFIGURATION
+# ============================================================================
+
+
+def get_database_url() -> str:
+    """
+    Extract database URL from environment and convert to async format.
+
+    Prisma URL: postgresql://user:pass@host:port/db?schema=platform&connect_timeout=60
+    Async URL:  postgresql+asyncpg://user:pass@host:port/db
+
+    Returns the async-compatible URL without query parameters (handled via connect_args).
+    """
+    prisma_url = Config().database_url
+
+    # Replace postgresql:// with postgresql+asyncpg://
+    async_url = prisma_url.replace("postgresql://", "postgresql+asyncpg://")
+
+    # Remove ALL query parameters (schema, connect_timeout, etc.)
+    # We'll handle these through connect_args instead
+    async_url = re.sub(r"\?.*$", "", async_url)
+
+    return async_url
+
+
+def get_database_schema() -> str:
+    """
+    Extract schema name from DATABASE_URL query parameter.
+
+    Returns 'platform' by default (matches Prisma configuration).
+    """
+    prisma_url = Config().database_url
+    match = re.search(r"schema=(\w+)", prisma_url)
+    return match.group(1) if match else "platform"
+
+
+# ============================================================================
+# ENGINE CREATION
+# ============================================================================
+
+
+def create_engine() -> AsyncEngine:
```

Comment:
For starter, can we use this on scheduler.py and compare its current implementation in there (in case we miss something here or vice versa) ?
This way we are sure the change is working and tested on an already running important component (which we already has been using sqlalchemy).